app/src/main/java/com/duckduckgo/app/browser/tabs/DefaultTabManager.kt

```diff
@@ -76,15 +83,27 @@ class DefaultTabManager @Inject constructor(
         }
     }
 
+    @OptIn(FlowPreview::class)
     override suspend fun requestAndWaitForNewTab(): TabEntity = withContext(dispatchers.io()) {
         val tabId = openNewTab()
-        return@withContext tabRepository.flowTabs.transformWhile { result ->
-            result.firstOrNull { it.tabId == tabId }?.let { entity ->
-                emit(entity)
-                return@transformWhile true
+        return@withContext tabRepository.flowTabs
+            .transformWhile { result ->
+                result.firstOrNull { it.tabId == tabId }?.let { entity ->
+                    emit(entity)
+                    return@transformWhile false // stop after finding the tab
+                }
+                return@transformWhile true // continue looking if not found
             }
-            return@transformWhile false
-        }.first()
+            .timeout(NEW_TAB_CREATION_TIMEOUT_LIMIT.seconds)
+            .catch { e ->
+                if (e is TimeoutCancellationException) {
+                    // timeout expired and the new tab was not found
+                    throw IllegalStateException("A new tab failed to be created within 1 second")
+                } else {
+                    throw e
+                }
+            }
+            .firstOrNull() ?: throw IllegalStateException("Tabs flow completed before finding the new tab")
```

Comment:
I'm a bit confused / worried about some of these changes tbh. We're now throwing exceptions if it doesn't complete in time, is something catching that or is crashing what we want here? What happens if the device is just old/slow/overloaded and happens to exceed 1s?

Why aren't these changes being feature flagged?

Reply:
> throws an exception when a new tab is not created because it's an invalid state in the DB that I'm not sure how we should recover from

That's the main bit i'm concerned about; i'm not sure it's always telling us about an invalid state. IIRC, there was the possibility of the underlying repo refusing to add another tab with the requested ID but that from a user perspective there would be no obvious problems. I'll dig in a little bit more though as I could be wrong. lmk if you want to chat sync about it.

Reply:
>Why aren't these changes being feature flagged? 
> It is, this is only used by tab swiping.

Thanks, that wasn't obvious from the diff that this function is currently only called from call sites already gated with the feature flag.

Reply:
> are you happy with the scope of what will be disabled? e.g., does all of tab swiping get disabled?

Yes, because this change fixes the original crash that caused the feature to be disabled in the first place. The thrown exceptions are just extra for us so that we know some unexpected circumstance occurred, which I assumed was irrecoverable. 

Just for context -- this function is being called when a ViewPager2 adapter requests a new fragment for a tab (which happens synchronously). The adapter expects a fragment to be created so I'm not sure what should happen if we can't get a tab.

Reply:
I'm not sure, either tbh. But if it's suitable FF-ed, I'm ok with trying it.

Reply:
>we could increase the time limit if you feel it's too short

I feel like it might be, but not 100% confident on any value for this. If you were pressing me, I'd throw out `2s` but don't expect me to back that up with justification 😅. I'm ok with trying `1s` as well. Will defer to your judgement.